sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,87 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_classes_helper(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+    # fit
+    classifier.fit(X, y)
+
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        msg_shape="Predicted shape of `decision_function` mismatches the expected " \
+                  "shape: expected '%s', got '%s'"
+        if len(classes) == 2:
+            assert_equal(decision.shape, (n_samples, ), msg_shape % \
+                        ((n_samples, ), decision.shape))
+            dec_pred = (decision.ravel() > 0).astype(np.int)
+            dec_exp = classifier.classes_[dec_pred]
+            msg="Predicted class mismatch for classifier %r: " \
+                "expected '%s', got '%s' " % \
+                (classifier, ", ".join(map(str, dec_exp)), ", ".join(map(str, y_pred)))
+            assert_array_equal(dec_exp, y_pred, err_msg=msg)
+        elif getattr(classifier, 'decision_function_shape', 'ovr') == 'ovr':
+            assert_equal(decision.shape, (n_samples, len(classes)), msg_shape % \
+                        ((n_samples, len(classes)), decision.shape))
+            decision_y = np.argmax(decision, axis=1).astype(int)
```

Comment:
We might want to try trigger a tie by repeating the test with the following as the only training/test data.
```
X = [[0], [0], [0]]
y = [0, 1, 2]
```

Maybe this is too fussy?

Reply:
This seems to be to be too fussy. Since the features are identical the predicted labels are identical as well : y_pred: array(['one', 'one', 'one'] and the expected are: array(['one', 'two', 'three'])

Reply:
Yes, I suppose I meant to check that it gave the first class in all cases, but you're right... Don't worry about it I to